internal/resource/resourcetest/builder.go

```diff
@@ -121,24 +123,35 @@ func (b *resourceBuilder) Write(t T, client pbresource.ResourceServiceClient) *p
 
 	res := b.resource
 
-	rsp, err := client.Write(context.Background(), &pbresource.WriteRequest{
-		Resource: res,
-	})
+	var rsp *pbresource.WriteResponse
+	var err error
 
-	require.NoError(t, err)
+	// Retry any writes where the error is a UID mismatch and the UID was not specified. This is indicative
+	// of using a follower to rewrite an object who is not perfectly in-sync with the leader.
+	retry.Run(t, func(r *retry.R) {
+		rsp, err = client.Write(context.Background(), &pbresource.WriteRequest{
```

Comment:
I can attempt but it will mean that our retry.R type needs to gain support for Cleanup. These funcs are using interfaces for the testing types so that it could handle either a testing.T, retry.R etc.

Arguably retry.R ought to support this anyways.

Reply:
Ah, gotcha. That makes sense. Alternatively, you could pass `t` here, which is probably better than nothing 🤷🏻

Reply:
I am seeing how hard it is to add cleanups to retry.R and then updating the interface to always have a Cleanup method

Reply:
I implemented cleanup functionality for retry.R and have refactored things to use TestContext everywhere.